src/Microsoft.Framework.Runtime/NuGet/Authoring/ManifestReader.cs

```diff
@@ -25,18 +26,15 @@ public static Manifest ReadManifest(XDocument document)
                     String.Format(CultureInfo.CurrentCulture, NuGetResources.Manifest_RequiredElementMissing, "metadata"));
             }
 
-            return new Manifest
-            {
-                Metadata = ReadMetadata(metadataElement),
-                Files = ReadFilesList(document.Root.ElementsNoNamespace("files").FirstOrDefault())
-            };
+            return new Manifest(
+                ReadMetadata(metadataElement),
+                ReadFilesList(document.Root.ElementsNoNamespace("files").FirstOrDefault()));
         }
 
         private static ManifestMetadata ReadMetadata(XElement xElement)
         {
             var manifestMetadata = new ManifestMetadata();
-            manifestMetadata.DependencySets = new List<ManifestDependencySet>();
-            manifestMetadata.ReferenceSets = new List<ManifestReferenceSet>();
+
```

Comment:
Ln. 38 should just be:
`manifestMetadata.MinClientVersionString = (string)xElement.Attribute("minClientVersion");

The `GetOptionalAttributeValue` does not add any value - attributes don't inherit namespaces and the `(string)` explicit conversion operator already handle the case if the attribute is not there (i.e. returns null). 

I suggest that the `GetOptionalAttributeValue()` method should be removed.